src/dns/server.rs

```diff
@@ -102,32 +102,36 @@ impl Server {
         let handler = dns::handler::Handler::new(store.clone());
         let mut server = ServerFuture::new(handler);
         info!(
-            address=%addr,
+            address=%address,
             component="dns",
             "starting local DNS server",
         );
-        // Bind and register the TCP socket.
-        let tcp_listener = socket_factory
-            .tcp_bind(addr)
-            .map_err(|e| Error::Bind(addr, e))?;
-        // Save the bound address.
-        let tcp_addr = tcp_listener.local_addr();
-        server.register_listener(
-            tcp_listener.inner(),
-            Duration::from_secs(DEFAULT_TCP_REQUEST_TIMEOUT),
-        );
+        let mut tcp_addr = None;
+        let mut udp_addr = None;
+        for addr in address.into_iter() {
```

Comment:
**non-blocking** I think we could get the same result using map on the iter and then take last of the resulting iter so we don't need to use mutable variables. I am not sure it is worth doing because the code may be less understandable and we are doing more than mapping one type onto another.

Reply:
yeah I was thinking of that but seemed more complex without much gain

Reply:
lets leave it as is then